### DISCUSSION TITLE
path_abandon消息没有支持，这个后面会实现吗？

### DISCUSSION THREAD

**Discussion by sharewzy** [Tencent/tquic#334] [Ideas（新功能的建议和讨论）]
path_abandon消息没有支持，这个后面会实现吗？目前没办法使用abandon path功能。

**Comment by iyangsj** [comment]
@sharewzy [draft-ietf-quic-multipath](https://datatracker.ietf.org/doc/html/draft-ietf-quic-multipath)多路径方案还在持续演进中，后续会在适合时机支持。

当前可通过建立新连接并关闭旧连接（如达到路径上限），来规避这类问题。

**Reply by sharewzy** [reply]
> @sharewzy [draft-ietf-quic-multipath](https://datatracker.ietf.org/doc/html/draft-ietf-quic-multipath)多路径方案还在持续演进中，后续会在适合时机支持。
> 
> 当前可通过建立新连接并关闭旧连接（如达到路径上限），来规避这类问题。

－－－》建立新连接关闭旧连接，是指重新执行quic_endpoint_connect 相关函数吗？没太明白这个规避逻辑。
另外看协议要求：发送path abandon消息之后，还需要执行retire connection id流程，如果想要先实现一个简单流程的话，是否只处理abandon消息就可以，接受消息之后释放掉相关path，space，dcid资源。

**Reply by iyangsj** [reply]
> 执行quic_endpoint_connect 相关函数吗

是的

> 另外看协议要求：发送path abandon消息之后，还需要执行retire connection id流程，如果想要先实现一个简单流程的话，是否只处理abandon消息就可以，接受消息之后释放掉相关path，space，dcid资源。

是的，最新草案已简化，但路径关闭仍有一些corner case需处理

**Reply by sharewzy** [reply]
> > 执行quic_endpoint_connect 相关函数吗
> 
> 是的
> 
> > 另外看协议要求：发送path abandon消息之后，还需要执行retire connection id流程，如果想要先实现一个简单流程的话，是否只处理abandon消息就可以，接受消息之后释放掉相关path，space，dcid资源。
> 
> 是的，最新草案已简化，但路径关闭仍有一些corner case需处理
－－－－》Corner case这部分可能会涉及到那些场景，能帮忙给些建议吗？感谢

**Reply by iyangsj** [reply]
参见 https://datatracker.ietf.org/doc/html/draft-ietf-quic-multipath#name-path-close 第3节讨论
